service/history/historyEventNotifier.go

```diff
@@ -60,8 +60,15 @@ type (
 
 var _ historyEventNotifier = (*historyEventNotifierImpl)(nil)
 
-func newHistoryEventNotification(domainID string, workflowExecution *gen.WorkflowExecution,
-	lastFirstEventID int64, nextEventID int64, previousStartedEventID int64, isWorkflowRunning bool) *historyEventNotification {
+func newHistoryEventNotification(
+	domainID string,
+	workflowExecution *gen.WorkflowExecution,
+	lastFirstEventID int64,
+	nextEventID int64,
+	previousStartedEventID int64,
+	isWorkflowRunning bool,
```

Comment:
since you are returning closeStatus as additional attr, could you also change isWorkflowRunning to workflow state, i.e. created / running / completed

Reply:
Sure. and the getMutableState history API will also return workflow state instead of isWorkflowRunning?

Reply:
depends, if you want to make the change, then it requires 2 release
1. return both isWorkflowRunning and workflowState
2. remove isWorkflowRunning